### PR TITLE
Move analytics script to a separate file

### PR DESCRIPTION
**Pull request by prtksxna**

### CODE REVIEW

**Comment by prtksxna** [comment]
I think introducing linting for a single file that is technically vendor code might be overkill. 

Happy to consider it though if someone feels differently :)

**Comment by prtksxna** [comment]
Maybe I am understanding this wrong. When we had added the script directly on the page we weren't deferring the loading of that script (it was already there). This keeps the behavior the same, doesn't it?

**Comment by Volker-E** [comment]
@prtksxna What does the current inline code do? It 
- adds a few global variables &
- adds an async/deferred script tag on top before it's calling element

Why do we put all this in a synchronous, blocking external script file call now?
If we stay with external file (see below), we need to add `async` and `defer` tag to the `script` reference tag.

Downloading external file is still a new HTTP request and I'm not sure if it's worth the maintenance win, but would go with majority here.

**Comment by prtksxna** [comment]
@Volker-E I agree, with the extra HTTP call we are making the user wait longer to see something useful. `async` will make the situation better, do you think we should also put the script at the end of the `body` so that it doesn't block HTML rendering?

**Comment by Volker-E** [comment]
@prtksxna The `script` tag referencing the script with `src` attribute or the script between a `script` tag. As I've stated above, and there's a ton of articles on `async` and `defer`, with asynchronous scripts we are already not blocking HTML rendering. 
The only reason I see currently for putting it external is @Ladsgroup comment about better maintainability. But it doesn't sell it to me if we loose visitor data and/or make script loading behavior worse.

File: design-principles.html [hunk 1]
```diff
@@ -20,26 +20,7 @@
...
-		}
-		if (window.addEventListener) {
-		    window.addEventListener('load', embedTrackingCode, false);
-		} else if (window.attachEvent) {
-		    window.attachEvent('onload',embedTrackingCode);
-		} else {
-		    embedTrackingCode();
-		}
-	</script>
+	<script src="js/analytics.js"></script>
```
Comment (Volker-E) [inline design-principles.html]: attrs missing

File: js/analytics.js [hunk 2]
```diff
@@ -0,0 +1,18 @@
+var _paq = _paq || [];
```
Comment (Volker-E) [inline js/analytics.js]: Let's follow MW coding conventions for whitespace and indentation. I'm clear that this hasn't been fully the case for the inline style, but ohay. ;)

**Comment by Volker-E** [comment]
I should have be clearer about my concern/ideated way of putting this external. The extra code for [integrating it after the `load` event](https://matomo.org/blog/2017/04/different-ways-embedding-piwik-tracking-code-faster-website-performance/) changes the situation a bit, what I've thought to end up with was:
```
<script src="js/analytics.js"></script>
```
holding:
```lang=js
var _paq = _paq || [];
_paq.push(['setDomains', ['*.design.wikimedia.org.']]);
_paq.push(['trackPageView']);
_paq.push(['enableLinkTracking']);
_paq.push(['setTrackerUrl', 'https://piwik.wikimedia.org/piwik.php']);
_paq.push(['setSiteId', '16']);
```
and 
```
<script src="https://piwik.wikimedia.org/piwik.js" async defer></script>
```

With `window.addEventListener` we'll need dynamic loading, so this is not an option. 
The async/defer attributes on your current `<script src="js/analytics.js"></script>` make no sense/difference then.

**Comment by Volker-E** [comment]
@prtksxna Should we proceed further on this, or abandon it?

**Comment by Volker-E** [comment]
The idea behind this patch has been integrated into and extended to #143. Respectively closing.

**Pull request closed without merging**